### PR TITLE
[issue-391] update readme to reflect the new usage after refactoring

### LINKED ISSUE #391: Update README in refactor branch
With infos regarding the current state and future development goals.

### PR DESCRIPTION
**Pull request by armintaenzertng**
fixes #391

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,65 +21,116 @@ This library implements SPDX parsers, convertors, validators and handlers in Pyt
...
-As always you should work in a virtualenv (venv). You can install a local clone
-of this repo with `yourenv/bin/pip install .` or install it from PyPI with
-`yourenv/bin/pip install spdx-tools`. Note that on Windows it would be `Scripts`
-instead of `bin`.
+## Library usage:
+1. **DATA MODEL**
+  * The `src.spdx.model` package constitutes the internal SPDX v2.3 data model (v2.2 is a simply a subset of this).
+  * SPDX objects are implemented via `@dataclass_with_properties`, a custom extension of `@dataclass`.
+    * Each class starts with a list of its properties and their possible types. When no default value is provided, the property is mandatory and must be set during initialization.
+    * Type checking is enforced from the type hints when initializing a new instance or setting/getting a property on an instance
```
Comment (meretp) [inline README.md:76]:
```suggestion
    * Type checking is enforced by the type hints when initializing a new instance or setting/getting a property on an instance
```

File: README.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ CI status (Linux, macOS and Windows): [![Install and Test][1]][2]
 # CURRENT STATE
 
 A major refactoring of large parts of the codebase is currently in progress. It is expected that functionality on `main`
-is severely limited during this process. Please check out
+is limited during this process. Please check out
```
Comment (meretp) [inline README.md:13]: Given the fact that we are now at a state where the functionality isn't really limited (probably except for the tag-value parser), I am wondering if we should enforce users to create issues here. This statement could make users think that all errors or misfunctionality is expected. What do you think?

**Review by meretp — CHANGES_REQUESTED** [review]
Thanks for updating this, I especially like the examples and the description for the library usages, I think this makes adaption much easier. I am not really satisfied with the paragraph to the current state, see my comment below.

File: README.md (hunk 1 above)
Comment (armintaenzertng) [reply README.md:76]: I will go with "Using the type hints, type checking is enforced when ..."

File: README.md [hunk 3]
```diff
@@ -21,65 +21,116 @@ This library implements SPDX parsers, convertors, validators and handlers in Pyt
...
 # Dependencies
 
 * PyYAML: https://pypi.org/project/PyYAML/ for handling YAML.
 * xmltodict: https://pypi.org/project/xmltodict/ for handling XML.
-* click: https://pypi.org/project/click/ for creating the CLI interface.
 * rdflib: https://pypi.python.org/pypi/rdflib/ for handling RDF.
-* typeguard: https://pypi.org/project/typeguard/ for using typehints.
+* click: https://pypi.org/project/click/ for creating the CLI interface.
+* typeguard: https://pypi.org/project/typeguard/ for type checking.
 * uritools: https://pypi.org/project/uritools/ for validation of URIs.
```
[the remark this answers is not in the document]
Comment (armintaenzertng) [reply README.md:134]: done

File: README.md [hunk 4]
```diff
@@ -21,65 +21,116 @@ This library implements SPDX parsers, convertors, validators and handlers in Pyt
...
+* Parse, convert, create and validate SPDX files
+* supported formats: Tag/Value, RDF, JSON, YAML, XML
 
-### Known Limitations
+# Planned features
 
-* No full 2.3 support for RDF format [#295](https://github.com/spdx/tools-python/issues/295)
-* No full license expression support [#10](https://github.com/spdx/tools-python/issues/10)
-* Output of the CLI parser is incomplete [#268](https://github.com/spdx/tools-python/issues/268)
+* up-to-date support of SPDX v3.0 as soon as it releases
```
[the remark this answers is not in the document]
Comment (armintaenzertng) [reply README.md:37]: done

File: README.md [hunk 5]
```diff
@@ -21,65 +21,116 @@ This library implements SPDX parsers, convertors, validators and handlers in Pyt
...
 
-* Include specialized validation for SPDX v2.2.1(ISO 5962:2021)
+As always you should work in a virtualenv (venv). You can install a local clone
+of this repo with `yourenv/bin/pip install .` or install it from PyPI with
+`yourenv/bin/pip install spdx-tools`. Note that on Windows it would be `Scripts`
+instead of `bin`.
 
 # How to use
 
 ## Command-line usage:
```
[the remark this answers is not in the document]
Comment (armintaenzertng) [reply README.md:48]: done

File: README.md [hunk 6]
```diff
@@ -21,65 +21,116 @@ This library implements SPDX parsers, convertors, validators and handlers in Pyt
...
+  
 * For help use `pyspdxtools --help`
 
-# Installation
-
-As always you should work in a virtualenv (venv). You can install a local clone
-of this repo with `yourenv/bin/pip install .` or install it from PyPI with
-`yourenv/bin/pip install spdx-tools`. Note that on Windows it would be `Scripts`
-instead of `bin`.
+## Library usage:
```
[the remark this answers is not in the document]
Comment (armintaenzertng) [reply README.md:71]: done

File: README.md (hunk 2 above)
Comment (armintaenzertng) [reply README.md:13]: We should rather encourage than enforce them, but I agree :D

Comment (armintaenzertng) [reply README.md]: Done, please check the new version! :)

File: README.md [hunk 7]
```diff
@@ -21,66 +23,118 @@ This library implements SPDX parsers, convertors, validators and handlers in Pyt
...
 
 * PyYAML: https://pypi.org/project/PyYAML/ for handling YAML.
 * xmltodict: https://pypi.org/project/xmltodict/ for handling XML.
-* click: https://pypi.org/project/click/ for creating the CLI interface.
 * rdflib: https://pypi.python.org/pypi/rdflib/ for handling RDF.
-* typeguard: https://pypi.org/project/typeguard/ for using typehints.
+* click: https://pypi.org/project/click/ for creating the CLI interface.
+* typeguard: https://pypi.org/project/typeguard/ for type checking.
 * uritools: https://pypi.org/project/uritools/ for validation of URIs.
+* license_expression: https://github.com/nexB/license-expression for handling SPDX license expressions.
```
Comment (meretp) [inline README.md:137]: for consistency we could use the pypi link here

**Review by meretp — APPROVED** [review]
LGTM

**Pull request merged**